### PR TITLE
Setup secrets boilerplate code from templates

### LINKED ISSUE #2606: Setup boilerplate code for secrets

### PR DESCRIPTION
**Pull request by gearama**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

closes #2606

### CODE REVIEW

**Review by vhvb1989 — APPROVED** [review]
Take a look for my suggestions and remove the dependency to Key Vault Common.  Then LGTM

File: sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <string>
+namespace Azure { namespace Security { namespace KeyVault { namespace Secrets {
+
+  class SecretClient final {
+  public:
+    std::string ClientVersion() const;
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp:11]: the api was added as part of the template, i'll remove it later on , for the moment it's used by the one and only unit test :)

File: sdk/keyvault/azure-security-keyvault-secrets/test/ut/secret_client_test.cpp [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <gtest/gtest.h>
+
+#include <azure/keyvault/secrets/secret_client.hpp>
+
+using namespace Azure::Security::KeyVault::Secrets;
+
+TEST(SecretClient, Basic)
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-secrets/test/ut/secret_client_test.cpp:10]: it's just for initial setup , to have one UT

File: sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+add_library(Azure::azure-security-keyvault-secrets ALIAS azure-security-keyvault-secrets)
+
+target_include_directories(
+  azure-security-keyvault-secrets
+    PUBLIC
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/inc>
+      $<INSTALL_INTERFACE:include>
+)
+
+target_link_libraries(azure-security-keyvault-secrets PUBLIC Azure::azure-security-keyvault-secrets)
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-secrets/CMakeLists.txt:52]: removed

**Pull request merged**